addons/Text_Selection/src/presenter.js

```diff
@@ -1719,33 +1730,147 @@ function AddonText_Selection_create() {
+        var $spans = $(findSelectableElements($html));
+        $spans.each(function() {
+            var number = parseInt($(this).attr('number'), 10);
+            var isInCorrectArray = $.inArray(number, presenter.markers.markedCorrect) >= 0;
+            if (isInCorrectArray)
+                $(this).wrap(`<u class="${CSS_CLASSES.PRINTABLE_CORRECT_ANSWER}"></u>`);
+        })
+    }
+
+    function upgradeHTMLForPrintableWhenShowUserAnswersPrintableStateMode($html) {
+        var $spans = $(findSelectedElements($html));
+        $spans.each(function() {
+            var $wrapper = $('<u></u>');
+            $wrapper.addClass(CSS_CLASSES.PRINTABLE_SELECTED);
+            $(this).wrap($wrapper);
+        })
+    }
+
+    function upgradeHTMLForPrintableWhenCheckAnswersPrintableStateMode($html) {
+        var $spans = $(findSelectedElements($html));
+        $spans.each(function() {
+            var $span = $( this );
+            var $wrapper = $('<u></u>');
+
+            $wrapper.addClass(CSS_CLASSES.PRINTABLE_SELECTED);
+            if (isCorrect($span)) {
+                $wrapper.addClass(CSS_CLASSES.PRINTABLE_CORRECT);
             } else {
-                content = content.replace(/\\correct{(.*?)}/g, '<strong>$1</strong>');
+                $wrapper.addClass(CSS_CLASSES.PRINTABLE_WRONG);
             }
-            content = content.replace(/\\wrong{(.*?)}/g, '<strong>$1</strong>');
-        }
-        $view.html(content);
+            $span.wrap($wrapper);
+        })
+    }
 
-        return $view[0].outerHTML;
-    };
+    function upgradeHTMLForPrintableElementsAccordingToMode($html, mode) {
+        if (mode === 'MARK_PHRASES')
+            upgradeHTMLForPrintableWhenMarkPhrasesToSelectMode($html);
+    }
+
+    function upgradeHTMLForPrintableWhenMarkPhrasesToSelectMode($html) {
+        findSelectableElements($html).wrap('<strong></strong>');
+    }
+
+    function cleanHTMLStructureFromNotPrintableAttributesAndClasses($html) {
+        findSelectableElements($html).removeClass(CSS_CLASSES.SELECTABLE);
+        findSelectedElements($html).removeClass(CSS_CLASSES.SELECTED);
+        findSpaceElements($html).contents().unwrap('<span></span>');
+        findNumberElements($html).contents().unwrap('<span></span>');
+    }
+
+    function findSelectedElements($html) {
+        return $html.find(`.${CSS_CLASSES.SELECTED}`);
+    }
+
+    function findSelectableElements($html) {
+        return $html.find(`.${CSS_CLASSES.SELECTABLE}`);
+    }
+
+    function findSpaceElements($html) {
+        return $html.find('span[left]');
+    }
+
+    function findNumberElements($html) {
+        return $html.find('span[number]');
+    }
+
+    function createViewForPrintable(model, $html) {
+        var $view = $("<div></div>");
+        $view.attr("id", model.ID);
+        $view.addClass(CSS_CLASSES.PRINTABLE);
+        $view.css("max-width", model["Width"] + "px");
+        $view.html($html.html());
```

Comment:
nie zwróciłem na to wcześniej uwagi, ale jesteś pewien, że to ma być $html.html()? To ci zwróci innerHTML. Jeżeli powinno być outerHTML, to zamiast  tego daj $html[0].outerHTML .

Reply:
Jak teraz patrzę to mógłbym to wyciągnięcie dać w createHTMLStructureForPrintable aby była lepsza przejrzystość